src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -60,7 +61,7 @@ public function buildForm(FormBuilderInterface $builder, array $options)
         $choiceList = $this->createChoiceList($options);
         $builder->setAttribute('choice_list', $choiceList);
 
-        if ($options['expanded']) {
+        if ($options['expanded'] && !in_array($options['widget'], array('text', 'hidden'))) {
             $builder->setDataMapper($options['multiple']
```

Comment:
Could be `if ('expanded' === $options['widget'])`

What is the point of having `checkbox` and `radio` value for `widget` option if you still have to deal with `multiple` option ?

`widget` might have few options like `text`, `hidden`, `select` and `expanded`